base_layer/core/src/transactions/transaction_components/payment_id.rs

```diff
@@ -121,7 +124,20 @@ impl Display for TxType {
 }
 
 #[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Default)]
-pub enum PaymentId {
+pub struct PaymentId {
+    inner: InnerPaymentId,
+}
+
+impl Deref for PaymentId {
+    type Target = InnerPaymentId;
+
+    fn deref(&self) -> &Self::Target {
+        &self.inner
+    }
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize, PartialEq, Eq, Default)]
+pub enum InnerPaymentId {
```

Comment:
```suggestion
enum InnerMemoField {
```
Dont expose this at all